firedrake/solving_utils.py

```diff
@@ -146,8 +146,28 @@ def __init__(self, problem, mat_type, pmat_type, appctx=None,
     @property
     def transfer_manager(self):
         if self._transfer_manager is None:
-            from firedrake import TransferManager
-            self._transfer_manager = TransferManager(use_averaging=True)
+            opts = PETSc.Options()
+            prefix = self.options_prefix or ""
+            if opts.hasName(prefix + "mg_transfer_manager"):
+                managername = opts[prefix + "mg_transfer_manager"]
+            elif opts.hasName(prefix + "fas_transfer_manager"):
+                managername = opts[prefix + "fas_transfer_manager"]
+            else:
+                managername = None
```

Comment:
Can you add a docstring here explaining the options? I also wonder in what circumstances the options prefix is `None` at this point (I think it is never `None` if the `_SNESContext` was built by `NLVS`).

Reply:
Docstring added.